File: RemoveDuplicateII_80.java
class Solution {
    public static int removeDuplicates(int[] nums) {
        int countNumber = 0;
        int savedNumber = nums[0];
        int k = 0;
        for(int i = 1; i < nums.length;i++) {
            if(nums[i] == savedNumber) {
                countNumber++;
            } else {
                countNumber = 0;
                savedNumber = nums[i];
            }
            
            if(countNumber < 2) {
                nums[++k] = nums[i];
            }
        }
        return k + 1;
        
//        int curr = 2;
//        for(int i = 0; i < nums.length; i++) {
//            if(nums[i] != nums[curr-2]) {
//                nums[curr++] = nums[i];
//            }
//        }
//        return curr;
    }
    
    public static void main(String args[]) {
        int[] nums = {0,0,1,1,1,1,2,3,3};
        System.out.println(removeDuplicates(nums));
        
        for(int i = 0; i < nums.length; i++) {
            System.out.println(nums[i] + "");
        }
    }
}

File: RemoveDuplicateII_80.py
class Solution(object):
    def removeDuplicates(self, nums):
        # k = 0
        # compareValue = -99999
        # count = 0
        # for num in nums :
        #     if num == compareValue :
        #         count += 1
        #         if count > 2:
        #             continue
        #     elif num != compareValue :
        #         count = 1
        #         compareValue = num
        #     nums[k] = num
        #     k += 1

        if len(nums) <= 2:
            return len(nums)

        k = 2
        for i in range(2, len(nums)) :
            if nums[i] != nums[k-2] :
                nums[k] = nums[i]
                k += 1
        return k


if __name__ == '__main__':
    nums = [0,0,1,1,1,1,2,3,3]
    solu = Solution()
    print(solu.removeDuplicates(nums))